manifests/zone.pp

```diff
@@ -13,6 +13,7 @@
   $zone_type = 'master',
   $allow_transfer = [],
   $allow_forwarder = [],
+  $allow_update = undef,
```

Comment:
It looks like we haven't been adding docstrings to this type.

Can you start with
```
# == Parameters
#
# [*allow_update*]
# ...
#
define dns::zone...
```